incentivization.md

```diff
@@ -52,231 +47,186 @@ Early P2P file-sharing networks employed reputation-based approaches and sticky
+2. If the client agrees, it pays and sends a proof of payment to the server.
+3. If the client does not agree, it sends a rejection message to the server.
+4. If the server receives a valid proof of payment before a certain timeout, it sends the response to the client.
+5. If the server receives a rejection message, or receives no message before a timeout, the server assumes that the client has rejected the offer.
 
 Potential issues:
-- A malicious client overwhelms a server with requests and doesn't follow through. A countermeasure: ignore requests from the same client if they come too often.
-- (other attacks?)
+- The client overwhelms a server with requests but doesn't proceed with payment. Countermeasure: ignore requests from the same client if they come too often.
+- The server and the client have no means to negotiate the price - see a later section on price negotiation.
 
 ### Proof of payment
 
 If the client agrees to the price, it sends a _proof of payment_ to the server.
 The nature of such proof depends on the means of payment.
-Assuming the payment takes place on a blockchain, it could simply be a transaction hash.
+Assuming the payment takes place on a blockchain, it could simply be a transaction hash (`txid`).
 
-It's unclear if we need to ensure that a particular txid is linked to a particular request.
+It's unclear whether we need to ensure that a given `txid` is linked to a particular request.
 Including request ID into the payment (a-la "memo field") threatens privacy.
-Not including it looks fine though, assuming the server keeps track which transactions it had received correspond to which requests (and responses).
-
-TODO: explore the idea of service credentials:
-- [https://forum.vac.dev/t/vac-sustainability-and-business-workshop/116](https://forum.vac.dev/t/vac-sustainability-and-business-workshop/116 "https://forum.vac.dev/t/vac-sustainability-and-business-workshop/116")
-- [https://github.com/vacp2p/research/issues/99](https://github.com/vacp2p/research/issues/99 "https://github.com/vacp2p/research/issues/99") 
-- [https://github.com/vacp2p/research/issues/135](https://github.com/vacp2p/research/issues/135 "https://github.com/vacp2p/research/issues/135")
+Not including it could lead to the server's confusion regarding which received payments correspond to which requests.
 
 #### Who pays first?
 
 We have to make a design decision: who pays first?
 Our options are:
-1. the client pays first and trust the server to deliver;
-2. the client pays after the fact: the server trusts the client;
+1. the client pays first;
+2. the client pays after the fac;
 3. the client pays partly upfront and partly after the fact;
-4. there is a third party (escrow) that ensures atomicity (a trusted third party or a semi-trusted, semi-automated entity like a smart contract).
+4. a third party (escrow) ensures atomicity (it may be a centralized trusted third party or a semi-trusted entity like a smart contract).
```

Comment:
Filecoin is using payment channels for retrieval but, ultimately, a similar problem. Might be interesting to explore: [https://spec.filecoin.io/systems/filecoin_token/payment_channels/](https://spec.filecoin.io/systems/filecoin_token/payment_channels/)

Reply:
Agree. See issue https://github.com/waku-org/research/issues/34.